xline-client/src/lib.rs

```diff
@@ -0,0 +1,417 @@
+    clippy::format_push_string,
+    clippy::iter_on_empty_collections,
+    clippy::iter_on_single_items,
+    clippy::large_include_file,
+    clippy::manual_clamp,
+    clippy::suspicious_xor_used_as_pow,
+    clippy::unnecessary_safety_comment,
+    clippy::unnecessary_safety_doc,
+    clippy::unused_peekable,
+    clippy::unused_rounding
+)]
+#![allow(
+    clippy::multiple_crate_versions, // caused by the dependency, can't be fixed
+    clippy::module_name_repetitions, // It will be more easy to use for the type name prefixed by module name
+)]
+#![cfg_attr(
+    test,
+    allow(
+        clippy::indexing_slicing,
+        unused_results,
+        clippy::unwrap_used,
+        clippy::as_conversions,
+        clippy::shadow_unrelated,
+        clippy::integer_arithmetic
+    )
+)]
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use curp::client::Client as CurpClient;
+use http::{header::AUTHORIZATION, HeaderValue, Request};
+use tonic::transport::Channel;
+use tower::Service;
+use utils::config::ClientTimeout;
+
+use crate::{
+    clients::{
+        auth::AuthClient, cluster::ClusterClient, election::ElectionClient, kv::KvClient,
+        lease::LeaseClient, lock::LockClient, maintenance::MaintenanceClient, watch::WatchClient,
+    },
+    error::{ClientError, Result},
+};
+
+/// Clients
+pub mod clients;
+/// Error.
+pub mod error;
+
+/// Xline client
+pub struct Client {
+    /// Name of the client
+    name: String,
+    /// kv client
+    kv: KvClient,
+    /// lease client
+    lease: LeaseClient,
+    /// lock client
+    lock: LockClient,
+    /// auth clinet
+    auth: AuthClient,
+    /// maintenance client
+    maintenance: MaintenanceClient,
+    /// watch client
+    watch: WatchClient,
+    /// cluster client
+    cluster: ClusterClient,
+    /// election client
+    election: ElectionClient,
+}
+
+impl Debug for Client {
+    #[inline]
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Client")
+            .field("name", &self.name)
+            .field("kv", &self.kv)
+            .field("lease", &self.lease)
+            .field("lock", &self.lock)
+            .field("auth", &self.auth)
+            .field("maintenance", &self.maintenance)
+            .field("watch", &self.watch)
+            .field("cluster", &self.cluster)
+            .field("election", &self.election)
+            .finish()
+    }
+}
+
+impl Client {
+    /// New `Client`
+    ///
+    /// # Errors
+    ///
+    /// If `StandaloneClient::connect` fails.
+    #[inline]
+    pub async fn connect(
+        all_members: HashMap<String, String>,
+        options: ClientOptions,
+    ) -> Result<Self> {
+        let name = String::from("client");
+        let channel = Self::build_channel(&all_members).await?;
+        let curp_client = Arc::new(CurpClient::new(None, all_members, options.curp_timeout).await);
+        let token = None;
```

Comment:
If this PR does not really use token, please add a todo

Reply:
done